pkg/driver/aci_driver.go

```diff
@@ -0,0 +1,760 @@
+		})
+	}
+	identity, err := d.getContainerIdentity(ctx, aciRG)
+	if err != nil {
+		return fmt.Errorf("Failed to get container Identity:%v", err)
+	}
+
+	_, err = d.createInstance(aciName, aciLocation, aciRG, op.Image, env, *identity)
+	if err != nil {
+		return fmt.Errorf("Error creating ACI Instance:%v", err)
+	}
+
+	// TODO: Check if ACR under ACI supports MSI
+	// TODO: Login to ACR if the registry is azurecr.io
+	// TODO: Add support for private registry
+
+	if d.deleteACIResources {
+		defer func() {
+			d.log("Deleting Container Instance")
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			containerGroupsClient := d.getContainerGroupsClient()
+			_, err := containerGroupsClient.Delete(ctx, aciRG, aciName)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to delete container error: %v\n", err)
+			}
+
+			d.log("Deleted Container ", aciName)
+		}()
+	}
+
+	// Check if the container is running
+	state, err := d.getContainerState(aciRG, aciName)
+	if err != nil {
+		return fmt.Errorf("Error getting container state :%v", err)
+	}
+
+	// Get the logs if the container failed immediately
+	if strings.Compare(state, "Failed") == 0 {
+		_, err := d.getContainerLogs(ctx, aciRG, aciName, 0)
+		if err != nil {
+			return fmt.Errorf("Error getting container logs :%v", err)
+		}
+
+		return errors.New("Container execution failed")
+	}
+
+	containerRunning := true
+	linesOutput := 0
+	for containerRunning {
+		d.log("Getting Container State")
+		state, err := d.getContainerState(aciRG, aciName)
+		if err != nil {
+			return fmt.Errorf("Error getting container state :%v", err)
+		}
+
+		if strings.Compare(state, "Running") == 0 {
+			linesOutput, err = d.getContainerLogs(ctx, aciRG, aciName, linesOutput)
+			if err != nil {
+				return fmt.Errorf("Error getting container logs :%v", err)
+			}
+
+			d.log("Sleeping")
+			time.Sleep(5 * time.Second)
+		} else {
+			if strings.Compare(state, "Succeeded") != 0 {
+				return fmt.Errorf("Unexpected Container Status:%s", state)
+			}
+
+			d.log("Container terminated successfully")
+			containerRunning = false
+		}
+
+	}
+
+	_, err = d.getContainerLogs(ctx, aciRG, aciName, linesOutput)
+	if err != nil {
+		return fmt.Errorf("Error getting container logs :%v", err)
+	}
+
+	d.log("Done")
+	return nil
+}
+
+// This will only work if the logs dont get truncated because of size.
+func (d *ACIDriver) getContainerLogs(ctx context.Context, aciRG string, aciName string, linesOutput int) (int, error) {
+	d.log("Getting Logs")
+	containerClient := d.getContainerClient()
+	logs, err := containerClient.ListLogs(ctx, aciRG, aciName, aciName, nil)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting container logs :%v", err)
+	}
+
+	lines := strings.Split(strings.TrimSuffix(*logs.Content, "\n"), "\n")
+	noOfLines := len(lines)
+	for currentLine := linesOutput; currentLine < noOfLines; currentLine++ {
+		fmt.Println(lines[currentLine])
```

Comment:
Should we rather pass the command writer rather than writing directly on stdout?